jobs/transforms/user_audits.py

```diff
@@ -39,3 +45,29 @@ def __init__(self):
         # self, like compiled re patterns, ARE LOST AFTER DESERIALIZATION!
         # https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled
         self._pattern = feconf.USER_ID_REGEX
+
+
+@audit_decorators.AuditsExisting(user_models.UserQueryModel)
+class ValidateOldModelsMarkedDeleted(beam.DoFn):
+    """DoFn to validate old models and mark them for deletion
+
+    Validate that there are no models that were last updated more than
+    four weeks ago, these models should be deleted.
+
+    """
+
+    def process(self, input_model):
+        """Function that checks if a model is old enough to mark them deleted.
+
+        Args:
+            input_model: user_models.UserQueryModel. Entity to validate.
+
+        Yields:
+            ModelExpiringError. An error class for expiring models.
+        """
+        model = job_utils.clone_model(input_model)
+        date_four_weeks_ago = (
```

Comment:
Let's not add the number in the variable name. Maybe `expiration_date` is better.

Reply:
done